module-4/backend/week_two.md

```diff
@@ -4,8 +4,59 @@ Fork this respository. Answer the questions to the best of your ability. Try to
 
 Note: When you're done, submit a PR. 
 
-TBD  
+1. What's one difference between ES5 and ES6?
+
+ES6 is moving JS to more natively support OOP.  One instance of this is the new Class structure.
+
+2. What's the difference between asynchronous and synchronous JavaScript? 
+
+synchonous JS runs all at once where as asynchronous allows us to tell the program to run Y line of code after X has happened successfully.
+
+3. What are the four pillars of Object Oriented programming?
+
+Abstraction - only show the outside world what is needed for it to interact with an object.  
+Polymorphism - can redefine functionality by reusing a name  
+Inheritance - create and share code through relationships between classes  
+Encapsulation - bind data and functions together in one class  
+
+4. What are some tools available in JavaScript to help you write object oriented code?
+
+`Class`es `static` methods and instance methods.
+
+5. What are some key concepts to be aware of when refactoring your JavaScript?
+
+Watch for code smells, follow SRP, watch for async problems/nested callbacks.
+
+6. What's a callback function and what are some reasons when we use/need callback functions?
+
+A function that gets called from another function when the first is finished executing.  It is a way of chaining actions to create asycronicity.
+
+7. What's the scope of variables in Javascript?
+
+Variables are scoped to their containing closure in JS.
+
+8. What's the difference between `==` and `===` in JavaScript?
+
+`===` checks type as well as equality.
+
+9. Why do front end frameworks exist?
+
+To make it easier to do things in JS.  What would take a line in jQuery might take 10 in vanilla JS.
```

Comment:
jQuery is better thought of as a library than a framework. Think about this question in the context of things like Ember, React (still a library, but closer to a framework), Angular, etc.